### PR TITLE
Add rustdoc for Latch.

### PR DESCRIPTION
**Pull request by weidagang**

### CODE REVIEW

File: src/storage/txn/latch.rs [hunk 1]
```diff
@@ -16,43 +16,64 @@ use std::collections::VecDeque;
 use std::hash::{Hash, SipHasher, Hasher};
 use std::usize;
 
+/// Latch which is used to serialize accesses to resources hashed to the same slot.
+///
+/// Latches are indexed by slot IDs. The keys of a command are hashed to slot IDs, then the command
+/// is added to the waiting queues of the latches.
+///
+/// If command A is ahead of command B in one latch, it must be ahead of command B in all the
+/// overlapping latches. This is an invariant ensured by the scheduler.
```
Comment (zhangjinpeng1987) [inline src/storage/txn/latch.rs]: this invariant is ensured by the process of gen_lock && acquire.

File: src/storage/txn/latch.rs [hunk 2]
```diff
@@ -16,43 +16,64 @@ use std::collections::VecDeque;
...
 
+    /// Returns true if all the required latches have be acquired, false otherwise.
     pub fn acquired(&self) -> bool {
         self.required_slots.len() == self.owned_count
     }
 }
 
+/// Latches which are used for concurrency control in the scheduler.
+///
+/// Each latch is index by a slot ID, hence the term latch and slot are used interchangably, but
```
Comment (siddontang) [inline src/storage/txn/latch.rs]: is indexed?

File: src/storage/txn/latch.rs [hunk 3]
```diff
@@ -16,43 +16,64 @@ use std::collections::VecDeque;
...
+/// conceptually a latch is a queue, and a slot is an index to a latch.
```
Comment (ngaut) [inline src/storage/txn/latch.rs]: s/a latch/the queue/

File: src/storage/txn/latch.rs (hunk 1 above)
Comment (weidagang) [inline src/storage/txn/latch.rs]: Done.

File: src/storage/txn/latch.rs (hunk 2 above)
Comment (weidagang) [inline src/storage/txn/latch.rs]: Done.

File: src/storage/txn/latch.rs (hunk 3 above)
Comment (weidagang) [inline src/storage/txn/latch.rs]: Done.

**Comment by ngaut** [comment]
LGTM

**Comment by ngaut** [comment]
Good job @weidagang

**Comment by weidagang** [comment]
Thanks, I'll keep adding docs as I'm getting familiar with the source code.

**Pull request merged**